pymongo/asynchronous/auth.py

```diff
@@ -177,13 +177,20 @@ def _auth_key(nonce: str, username: str, password: str) -> str:
     return md5hash.hexdigest()
 
 
-def _canonicalize_hostname(hostname: str) -> str:
+def _canonicalize_hostname(hostname: str, option: str | bool) -> str:
     """Canonicalize hostname following MIT-krb5 behavior."""
     # https://github.com/krb5/krb5/blob/d406afa363554097ac48646a29249c04f498c88e/src/util/k5test.py#L505-L520
+    if option in [False, "none"]:
+        return hostname
+
     af, socktype, proto, canonname, sockaddr = socket.getaddrinfo(
         hostname, None, 0, 0, socket.IPPROTO_TCP, socket.AI_CANONNAME
     )[0]
 
+    # For forward just to resolve the cname as dns.lookup() will not return it.
+    if option == "forward":
+        return canonname.lower()
```

Comment:
I'm still confused by this method. DRIVERS-1803 says:
> "forward" is equivalent to the current behaviour of true (a forward DNS lookup)

But here we are introducing new behavior for "forward". Is that because our behavior for "true" is actually "forwardAndReverse"? If so could you add a comment mentioning that?

Reply:
Ah, I misinterpreted the Node implementation and got confused.   I'll redo the logic.

Reply:
No, it was the description in the ticket that was incorrect.  I verified by reading the kerberos spec, the drivers spec, and the node implementation that this is the correct implementation.

Reply:
I also updated the ticket description